app/src/main/java/com/jakdor/apapp/ui/registration/RegistrationFragment.kt

```diff
@@ -0,0 +1,279 @@
+package com.jakdor.apapp.ui.registration
+
+import android.graphics.Color
+import android.os.Bundle
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.ViewModelProviders
+import com.jakdor.apapp.R
+import com.jakdor.apapp.di.InjectableFragment
+import kotlinx.android.synthetic.main.fragment_registration.*
+import javax.inject.Inject
+
+class RegistrationFragment : Fragment(), InjectableFragment {
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelProvider.Factory
+
+    var viewModel: RegistrationViewModel? = null
+
+    var isPasswordCorrect: Boolean = false
+    var isRePasswordCorrect: Boolean = false
+    var isEmailCorrect: Boolean = false
+    var isLoginCorrect: Boolean = false
+    var isNameNotEmpty: Boolean = false
+    var isSurnameNotEmpty: Boolean = false
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
+                              savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.fragment_registration, container, false)
+    }
+
+    override fun onActivityCreated(savedInstanceState: Bundle?) {
+        super.onActivityCreated(savedInstanceState)
+        if(viewModel == null) {
+            viewModel = ViewModelProviders.of(this, viewModelFactory).get(RegistrationViewModel::class.java)
+        }
+
+        val surname: String = surname_editText.text.toString()
+        surname_editText.error = viewModel?.isEmptyValidation(surname)
+
+        val name: String = name_editText.text.toString()
+        name_editText.error = viewModel?.isEmptyValidation(name)
+
+        val email: String = email_editText.text.toString()
+        email_editText.error = viewModel?.isEmptyValidation(email)
+
+        val login: String = login_editText.text.toString()
+        login_editText.error = viewModel?.isEmptyValidation(login)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        register_button.setOnClickListener {
+
+            val password: String = password_editText.text.toString()
+            val rePassword: String = rePassword_editText.text.toString()
+
+            if(password != rePassword){
+                rePassword_editText.error = "Podane hasła się nie zgadzają"
+            }
+        }
+
+        register_button.isEnabled = false
+
+        password_editText.addTextChangedListener(object : TextWatcher{
+            override fun afterTextChanged(s: Editable?) {
+                val password: String = password_editText.text.toString()
```

Comment:
W listenerach powinieneś tylko wywoływać funkcje ViewModelu, masz bardzo dużo logiki we Fragmencie, która spokojnie mogła by być tam